### DISCUSSION TITLE
Microsoft.Services.Store.SDK - Feedback Hub in WinUI3

### DISCUSSION THREAD

**Discussion by mvpete** [microsoft/WindowsAppSDK#2365] [Q&A]
I am trying to follow [this UWP article](https://docs.microsoft.com/en-us/windows/uwp/monetize/launch-feedback-hub-from-your-app) for launching feedback hub, but unable to resolve references. Is it possible to launch the feedback hub from a Win UI 3 application? I've installed the SDK, also tried NuGet packages with no luck. My app is C# .NET 6 application, on Windows 11, using Visual Studio 2022.

**Comment by castorix** [comment]
> I am trying to follow [this UWP article](https://docs.microsoft.com/en-us/windows/uwp/monetize/launch-feedback-hub-from-your-app) for launching feedback hub, but unable to resolve references. Is it possible to launch the feedback hub from a Win UI 3 application? I've installed the SDK, also tried NuGet packages with no luck. My app is C# .NET 6 application, on Windows 11, using Visual Studio 2022.

I could compile with [CsWinRT](https://github.com/microsoft/CsWinRT) package, with
` <TargetFramework> net5.0-windows10.0.19041.0 </TargetFramework>`
on Windows 10 21H1
But then it crashes at execution...
(even just with 
`Microsoft.Services.Store.Engagement.StoreServicesFeedbackLauncher.IsSupported())`
(I added a Manifest for the minimum Windows 10 version but nothing changed...)

**Reply by mvpete** [reply]
I didn't even get it to compile... I looked at launching it with the feedback URI, which works... but it's not targeted at the specific app. Though, I've not done much research down that road.

**Comment by JoeTomkinson** [comment]
Looks as though the package just doesn't work on .NET 6 WinUI apps. I've had a bit of a dig around and although it doesn't look like there are any glaringly obvious package dependency issues on the face of it there must be as the project refuses to allow the referencing of the package.

**Comment by karter16** [comment]
Has there been any progress on this. The lack of availability of features for the Services.Store.SDK for WindowsAppSDK apps is pretty disappointing.

**Reply by DarranRowe** [reply]
Honestly, without people giving concrete examples of why this doesn't work for them, it is difficult to provide help.
There are two reasons why this would fail from what I can see.
1) The application is unpackaged and you are using the Windows App SDK bootstrapper to load the Windows App SDK. IIRC, some of the older Windows App SDK bootstrapper/detours stuff would assume that everything under Microsoft.* was part of the Windows App SDK and would just fail. I could be wrong with this point.
2) This is written primarily as a C++ UWP component, thus it requires the UWP VC Runtime. Unpackaged desktop applications need to use the Microsoft.VCRTForwarders.140 NuGet package and have the latest VC 2015-2022 runtime installed. This places the *_app.dll versions of the VC Runtime in the application directory to satisfy all of the dependencies. Packaged applications would either need to use this NuGet package along with the desktop VC Runtime package, or add a reference to the UWP VC Runtime package in the manifest.

But this will activate in desktop applications without issue if you prepare things correctly. Yes, this also applies to WinUI 3 applications, since they run in the desktop application environment.

**Reply by mklemarczyk** [reply]
@DarranRowe Create a new WinUI 3 app, and add nuget package. You will see that Visual Studio does not recognize the namespace for class StoreServicesFeedbackLauncher. 
![image](https://github.com/user-attachments/assets/47772c97-3ab9-48e5-ae71-e7789a38ec6b)
![image](https://github.com/user-attachments/assets/03f3bdb9-ce4c-40bc-86e1-fd3c5f2bd9cb)

```xml
<Project Sdk="Microsoft.NET.Sdk">
	<PropertyGroup>
		<OutputType>WinExe</OutputType>
		<TargetFramework>net8.0-windows10.0.22621</TargetFramework>
		<TargetPlatformMinVersion>10.0.17763.0</TargetPlatformMinVersion>
		<ApplicationIcon>Assets/WindowIcon.ico</ApplicationIcon>
		<ApplicationManifest>app.manifest</ApplicationManifest>
		<Platforms>x86;x64;ARM64</Platforms>
		<RuntimeIdentifiers>win-x86;win-x64;win-arm64</RuntimeIdentifiers>
		<ImplicitUsings>enable</ImplicitUsings>
		<Nullable>enable</Nullable>
		<UseWinUI>true</UseWinUI>
	</PropertyGroup>
```

App packaged with separate Msix project.

Steps to reproduce : 
* Install Visual Studio 2022 with .NET 9.
* Create new project for WinUI 3.
* Add Store package `<PackageReference Include="Microsoft.Services.Store.Engagement" Version="10.2307.3001" />`
* try to compile with line : `Microsoft.Services.Store.Engagement.StoreServicesFeedbackLauncher.IsSupported()`

**Reply by DarranRowe** [reply]
@mklemarczyk 

In that case it is understandable. There is no CsWinRT projection. This is also technically outside of the scope of this repo.

**Reply by mklemarczyk** [reply]
@DarranRowe it works with C# in scope of UWP. So what is missing now for WindowsAppSDK & WinUI3 ? Who should manage support for it ? Microsoft don't want to support own products and technologies anymore ?

**Reply by DarranRowe** [reply]
It works with C# in scope of UWP because that should be based on the .NET framework or .NET core. [.NET 5 removed builtin support for WinRT](https://learn.microsoft.com/en-us/dotnet/core/compatibility/interop/5.0/built-in-support-for-winrt-removed), so nothing is missing from the Windows App SDK/WinUI 3, it is .NET which is missing everything.

This is the reason why the [CsWinRT project](https://github.com/microsoft/CsWinRT) exists, but the Store Services SDK team never added a projection to the Microsoft.Services.Store.Engagement package. So the team which are responsible for this would be the Store Services SDK team. Maybe you can go to the [NuGet page](https://www.nuget.org/packages/Microsoft.Services.Store.Engagement) and use contact owners there to try to get something done.

**Reply by mklemarczyk** [reply]
Where could we create a request for the missing projection @DarranRowe @Sergio0694 @mominshaikhdevs  ?
